### ISSUE TITLE
firebase integration stuck on loading project

### ISSUE THREAD

**Issue by alexw10** [stackblitz/core#759]
I clicked the sign in and allowed permissions for firebase intergration but it just keeps spinning here:
![image](https://user-images.githubusercontent.com/9453636/47740872-e6a20780-dc46-11e8-848b-f6cbda6aefaf.png)

Any ideas?

**Comment by Misiu** [comment]
I have the exact same issue
![stackblitz_firebase](https://user-images.githubusercontent.com/1741838/48612220-9df98680-e988-11e8-85e7-25776514e4f5.png)

**Comment by LeviBarker** [comment]
I'm experiencing the same thing, it happened when the login popup for Google came up, but wasn't completed before leaving the page.

**Comment by davidrensh** [comment]
same here. Google firebase team do not want to give permission or just pure integration issue? 👍

**Comment by richarddavenport** [comment]
I tried to remove stackblitz access to my account, I try to add it back, but the network shows a `403`. Anyway to re-authenticate?

**Comment by DQM** [comment]
Still spinning.... been 3 days now. Did all the remove access & clear cache process. Anyone able to fix this, kindly update

**Comment by Misiu** [comment]
@apai4 any news?

**Comment by Dellos7** [comment]
Same issue here. What it's funny is that I am able to deploy from one of my accounts, but when I try with other account that I've created some days ago, it just stucks in "Loading projects..."

**Comment by Dellos7** [comment]
I think I have resolved it.

You need to:
* Create a Firebase project.
* Create a database for the project.

It is **important** that you hit **Create database** and follow the steps.

Then, you'll have your projects listed in StackBlitz's Firebase integration.

![](https://i.imgur.com/dAtXc92.png)

**Comment by krishnanandreddy** [comment]
> I think I have resolved it.
> 
> You need to:
> 
> * Create a Firebase project.
> * Create a database for the project.
> 
> It is **important** that you hit **Create database** and follow the steps.
> 
> Then, you'll have your projects listed in StackBlitz's Firebase integration.
> 
> ![](https://camo.githubusercontent.com/fbcf93b1f663c9dca246a86ce51c4d04c1f7921b/68747470733a2f2f692e696d6775722e636f6d2f644174586339322e706e67)

It works after creating Database.
Thanks

**Comment by richarddavenport** [comment]
Just to follow up, this was working for me at one point, I remove stackblitz as an authorized application from my google account, now I'm trying to re-authenticate, but I just see the spinner.
<img width="892" alt="screen shot 2019-02-21 at 9 06 22 am" src="https://user-images.githubusercontent.com/3596012/53174588-299e3700-35b8-11e9-81a8-c2358d8d1f5f.png">

**Comment by lyang1810** [comment]
From the screenshot, the 403 comes from https://stackblitz.com/api/user/tokens/google, which relates to how it handles OAuth tokens. In the sequence of API calls, the error happens before any call is made to the Firebase Management API for listing Firebase projects, so it is likely an OAuth issue instead of a Firebase IAM permission issue.

However, the fix with "Create database" could indicate a separated issue with IAM permissions. Not sure about it yet.

**Comment by apai4** [comment]
@alexw10 @Misiu @richarddavenport I pushed a fix that prompts to sign in with Google when the 403 happens, can you see if this fixes the issue for you? 

I can also confirm that projects currently don't seem to show up until a database is created, as @Dellos7 discovered. I'll see if there's anything we can do to have them show up without having to do that manually.

**Comment by Misiu** [comment]
@apai4 I've just tried Your fix.
First I've created a database inside my target firebase project, then I opened one of my projects on Stackblitz. I had to log in to firebase a couple of times because popup stuck at some point.
Finally, after a couple of refreshes, I was able to deploy everything.

Ideally, the manual database creation step should be avoided but for me, one manual step isn't a problem.

**Comment by richarddavenport** [comment]
@apai4 Awesome! Thanks for doing that. Just to report back on my situation, I had to jiggle the handle to get it to work (wasn't smooth).

I was able to click the google login in link, so first step was done!
But when I connect my google account the login pop window logged me in and did not close:
<img width="1428" alt="screen shot 2019-02-27 at 3 21 13 pm" src="https://user-images.githubusercontent.com/3596012/53520520-d50d2700-3aa3-11e9-89d1-e1268e40833e.png">

I still had the google login button, like it didn't work.
I tried refreshing stackblitz, didn't help...
I tried signing out, signing back in, didn't help...
Opened stackblitz in incognito mode logged in, re-authenticated (again) and this time it worked. Maybe local storage/cache was messed up? Anyway, closed incognito mode, back to a regular browser, refreshed, and I'm all good now!

Thanks, hope this helps!
